casr/src/bin/casr-ubsan.rs

```diff
@@ -384,19 +391,25 @@ fn main() -> Result<()> {
 
     // Extract ubsan warnings
     info!("Extracting UBSAN warnings...");
-    info!("Using {} threads", num_of_threads);
-    let warnings: Vec<(&PathBuf, Vec<(UbsanWarning, CrashLine)>)> = custom_pool.install(|| {
-        inputs
-            .par_iter()
-            .filter_map(|input| {
-                let Ok(input_warnings) = extract_warnings(input, &argv, timeout) else {
-                    warn!("Failed to run program with input file {:?}", input);
-                    return None
-                };
-                Some((input, input_warnings))
-            })
-            .collect()
-    });
+    info!("Using {} threads", num_of_threads - 1);
+    let counter = RwLock::new(0_usize);
+    let total = inputs.len();
+    type Warning<'a> = (&'a PathBuf, Vec<(UbsanWarning, CrashLine)>);
+    let (warnings, _): (Vec<Warning>, _) = custom_pool.join(
+        || {
+            inputs
+                .par_iter()
+                .filter_map(|input| {
+                    let Ok(input_warnings) = extract_warnings(input, &argv, timeout, &counter) else {
+                        warn!("Failed to run program with input file {:?}", input);
+                        return None
+                    };
```

Comment:
We may increment counter just here. So, there is no need for additional `counter` argument for `extract_warnings`.

Reply:
Done